etcd.go

```diff
@@ -0,0 +1,127 @@
+package dsync
+
+import (
+	"github.com/coreos/etcd/client"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+	"sync"
+	"time"
+)
+
+type etcdMutex struct {
+	backoff BackoffFunc
+	ctx     context.Context
+	etcd    client.KeysAPI
+	key     string
+	locked  bool
+	m       sync.Mutex
+	refresh time.Duration
+	stop    chan struct{}
+	ttl     time.Duration
+	uuid    string
+	wg      sync.WaitGroup
+}
+
+func (e *etcdMutex) Lock() {
+	var i int64
```

Comment:
Don't declare this variable so far away from where it is being used. You want your variables to be as locally blocked as possible.